db/upgrade.php

```diff
@@ -12,250 +14,53 @@
-        $table->add_field('id', XMLDB_TYPE_INTEGER, '20', null, XMLDB_NOTNULL, XMLDB_SEQUENCE, null);
-        $table->add_field('course', XMLDB_TYPE_INTEGER, '20', null, XMLDB_NOTNULL, null, null);
-        $table->add_field('coursefeedbackid', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, null);
-        $table->add_field('questionid', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, null);
-        $table->add_field('textanswer', XMLDB_TYPE_TEXT, 'medium', null, null, null, null);
-        $table->add_field('timemodified', XMLDB_TYPE_INTEGER, '10', null, null, null, '0');
-
-        // Add keys.
-        $table->add_key('primary', XMLDB_KEY_PRIMARY, ['id']);
-        $table->add_key('coursefeedbackid', XMLDB_KEY_FOREIGN, ['coursefeedbackid'], 'block_coursefeedback', ['id']);
-        $table->add_key('courseid', XMLDB_KEY_FOREIGN, ['course'], 'course', ['id']);
-
-        // Conditionally create table.
-        if (!$dbman->table_exists($table)) {
-            $dbman->create_table($table);
-        }
-
-        upgrade_block_savepoint(true, 2024010900, 'coursefeedback');
-    }
-
-    if ($oldversion == 2024015000 ) {
-        // Following changes need to be checked only for version '2024015000'.
-
-        $dbman = $DB->get_manager();
-        $table = new xmldb_table('block_coursefeedback_answers');
-
-        // Drop 'textanswer' field in 'block_coursefeedback_answers'.
-        $field = new xmldb_field('textanswer', XMLDB_TYPE_TEXT, 'medium', null, null, null, null, 'answer');
-        $dbman->drop_field($table, $field);
-
-
-        // Check if the 'answer' field in the table 'block_coursefeedback_answers' has the correct 'NOTNULL' value.
-        $field = new xmldb_field('answer', XMLDB_TYPE_INTEGER, '4', null, XMLDB_NOTNULL, null, null, 'questionid');
-        // Drop index temporarily to make field change possible.
-        $index = new xmldb_index('bloccouransw_coucouque_ix', XMLDB_INDEX_NOTUNIQUE, ['course', 'coursefeedbackid', 'questionid', 'answer']);
-        if ($dbman->index_exists($table, $index)) {
-            $dbman->drop_index($table, $index);
+    // 2) Verify that the admin has accepted the major overhaul and possible data loss.
+    if ($oldversion < 2025050901) {
+        $confirmed = get_config('block_coursefeedback', 'confirmoverhaul');
+        if (empty($confirmed)) {
+            throw new moodle_exception(
+                'You must confirm the major overhaul and possible data loss '
+                . 'in the Course Feedback block settings before continuing.'
+            );
         }
 
-        // Make sure NOTNULL-Option of the 'answer' field is 'true'.
-        $dbman->change_field_notnull($table, $field);
+        // TODO: data archival/migration logic here.
 
-        // Add index again.
-        $dbman->add_index($table, $index);
-
-        upgrade_block_savepoint(true, 2024015001, 'coursefeedback');
+        upgrade_plugin_savepoint(true, 2025050901, 'block_coursefeedback');
```

Comment:
Die Funktion braucht bei mir block und coursefeedback als getrennte Argumente, aber das würde funktionieren.

```suggestion
        upgrade_block_savepoint(true, 2025050901, 'coursefeedback');
```